src/features/posts.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-param-reassign */
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const initPosts = createAsyncThunk('posts/fetch', (userId: number) => {
+  return getUserPosts(userId);
+});
```

Comment:
Consider adding error handling for the `getUserPosts` API call to manage potential failures gracefully. This will help in providing feedback to the user if the post fetching fails.